resource-providers/src/main/java/io/opentelemetry/resourceproviders/AppServerServiceNameProvider.java

```diff
@@ -37,11 +38,14 @@ public AppServerServiceNameProvider() {
   @Override
   public Resource createResource(ConfigProperties config) {
     String serviceName = detectServiceName();
-    if (serviceName != null) {
-      logger.log(INFO, "Auto-detected service name '{0}'.", serviceName);
-      return Resource.create(Attributes.of(SERVICE_NAME, serviceName));
+    if (serviceName == null) {
+      logger.log(
+          WARNING,
+          "Service name could not be detected using common application server strategies.");
+      return Resource.empty();
     }
-    return Resource.empty();
+    logger.log(INFO, "Auto-detected service name {0}.", serviceName);
```

Comment:
For the record, I think that `INFO` actually fits this log - you could add something mentioning where the service name was extracted from (e.g. `... in a app server deployment`)

Reply:
Ok, I kept that one at `INFO`.